pkg/cloud/cloud.go

```diff
@@ -102,6 +113,23 @@ func NewCloud() (Cloud, error) {
 	}, nil
 }
 
+// NewCloudWithRole returns a new instance of AWS cloud after assuming an aws role
+// It panics if session is invalid
```

Comment:
panic sounds wrong? if we don't have permission to assume the role it panics?

Reply:
Agreed, I'll update the comment accordingly